larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h

```diff
@@ -0,0 +1,319 @@
+     *
+     *  @param  features the input features
+     *
+     *  @return the classification probability
+     */
+    double CalculateProbability(const DoubleVector &features) const;
+
+    /**
+     *  @brief  Read the bdt parameters from an xml file
+     *
+     *  @param  bdtFileName the sml file name
+     *  @param  bdtName the name of the bdt
+     */
+    void ReadXmlFile(const std::string &bdtFileName, const std::string &bdtName);
+
+private:
+    class Node
+    {
+    public:
+        /**
+         *  @brief  Constructor
+         *  
+         *  @param  nodeID id of node
+         *  @param  parentNodeID id of parent node
+         *  @param  leftChildNodeID id of left child node
+         *  @param  rightChildNodeID id of right child node
+         *  @param  isLeaf is node a leaf
+         *  @param  threshold threshold for cut
+         *  @param  variableID index
+         *  @param  outcome, only written if leaf node 
+         */
+        Node(const int &nodeID, const int &parentNodeID, const int &leftChildNodeID, const int &rightChildNodeID, const bool &isLeaf, const double &threshold, const int &variableID, const bool outcome = false);
+
+        /**
+         *  @brief  Return node id
+         */
+        int GetNodeID() const;
+
+        /**
+         *  @brief  Return parent node id
+         */
+        int GetParentNodeID() const;
+
+        /**
+         *  @brief  Return left child node id
+         */
+        int GetLeftChildNodeID() const;
+
+        /**
+         *  @brief  Return right child node id
+         */
+        int GetRightChildNodeID() const;
+
+        /**
+         *  @brief  Return is the node a leaf
+         */
+        bool IsLeaf() const;
+
+        /**
+         *  @brief  Return node threshold
+         */
+        double GetThreshold() const;
+
+        /**
+         *  @brief  Return cut variable
+         */
+        int GetVariableID() const;
+
+        /**
+         *  @brief  Return outcome
+         */
+        bool GetOutcome() const;
+
+    private:
+        const int       m_nodeID;               ///< Node id
+        const int       m_parentNodeID;         ///< Parent node id
+        const int       m_leftChildNodeID;      ///< Left child node id
+        const int       m_rightChildNodeID;     ///< Right child node id
+        const bool      m_isLeaf;               ///< Is node a leaf
+        const double    m_threshold;            ///< Threshold used for decision if decision node
+        const int       m_variableID;          ///< Variable cut on for decision if decision node
+        const bool      m_outcome;              ///< Outcome if leaf node
+    };
+
+    typedef std::map<int, const Node*> IDToNodeMap;
+
+    class WeakClassifier
+    {
+    public: 
+        /**
+         *  @brief  Constructor, set hierarchy for nodes 
+         */
+        WeakClassifier(IDToNodeMap &idToNodeMap, const double &weight, const int &treeID);
```

Comment:
const `IDToNodeMap`? Just const by value for other two arguments.

Reply:
Done.